json/src/ser.rs

```diff
@@ -1245,11 +1288,11 @@ fn format_escaped_char<W: ?Sized, F: ?Sized>(wr: &mut W, formatter: &mut F, valu
     where W: io::Write,
           F: Formatter,
 {
-    // FIXME: this allocation is required in order to be compatible with stable
-    // rust, which doesn't support encoding a `char` into a stack buffer.
-    let mut s = String::new();
-    s.push(value);
-    format_escaped_str(wr, formatter, &s)
+    use std::io::Write;
+    let mut buf = [0; 4];
+    write!(&mut buf[..], "{}", value).unwrap();
+    let str = unsafe { str::from_utf8_unchecked(&buf[0..value.len_utf8()]) };
```

Comment:
Add a brief justification of why this is safe along with a link to <a href="https://github.com/serde-rs/json/issues/270"> https<span></span>://github.com/serde-rs/json/issues/270</a>.